gson/src/test/java/com/google/gson/GsonTest.java

```diff
@@ -368,4 +360,223 @@ public CustomClass3() {
       this(NO_ARG_CONSTRUCTOR_VALUE);
     }
   }
+
+  /**
+   * Verifies that {@link Gson#getAdapter(TypeToken)} does not put broken adapters
+   * into {@code typeTokenCache} when caller of nested {@code getAdapter} discards
+   * exception, e.g.:
+   *
+   * Field dependencies:
+   * ClassA
```

Comment:
Have removed this test again since it was only relevant for the contrived situation of discarding exceptions from `getAdapter`, which I have removed as well.